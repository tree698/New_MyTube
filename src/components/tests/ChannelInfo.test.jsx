import { render, screen, waitFor } from '@testing-library/react';
import { Route } from 'react-router-dom';
import { withAllContexts, withRouter } from '../../tests/utils';
import ChannelInfo from '../ChannelInfo';

describe('ChannelInfo', () => {
  const fakeYoutube = {
    channelImageURL: jest.fn(),
  };

  afterEach(() => fakeYoutube.channelImageURL.mockReset());

  it('renders correctly', async () => {
    const { asFragment } = renderChannelInfoWithCallback(() => 'url');

    await waitFor(() => screen.getByRole('img'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders without image', () => {
    renderChannelInfoWithCallback(() => {
      throw new Error('error');
    });

    expect(screen.queryByRole('img')).toBeNull();
  });

  it('renders with image', async () => {
    renderChannelInfoWithCallback(() => 'url');

    await waitFor(() => expect(screen.getByRole('img')).toBeInTheDocument());
  });

  function renderChannelInfoWithCallback(callback) {
    fakeYoutube.channelImageURL.mockImplementation(callback);
    return render(
      withAllContexts(
        withRouter(
          <Route path="/" element={<ChannelInfo id="id" name="channel" />} />
        ),
        fakeYoutube
      )
    );
  }
});
