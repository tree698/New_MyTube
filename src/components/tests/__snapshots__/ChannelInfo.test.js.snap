// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfo renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex my-4 mb-8 items-center"
  >
    <img
      alt="channel"
      class="w-14 h-14 rounded-full"
      src="url"
    />
    <p
      class="text-lg font-medium ml-2"
    >
      channel
    </p>
  </div>
</DocumentFragment>
`;
